src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResource.cs

```diff
@@ -537,68 +544,53 @@ ex.InnerException is IOException &&
                 }
             }
 
-            return null;
+            return [];
         }
 
-        private async Task<SortedDictionary<NuGetVersion, PackageInfo>> ConsumeFlatContainerIndexAsync(Stream stream, string id, string baseUri, CancellationToken token)
+        private static async Task<HashSet<NuGetVersion>> ConsumeFlatContainerIndexAsync(Stream stream, string id, string baseUri, CancellationToken token)
         {
-            var doc = await stream.AsJObjectAsync(token);
+            var json = await JsonSerializer.DeserializeAsync<FlatContainerVersionList>(stream, cancellationToken: token);
```

Comment:
>JsonSerializer.DeserializeAsync [](http://example.com/codeflow?start=29&length=31)

This shows up as a very small amount of the remaining allocations, so there is definitely no need to pursue the other PR.

Reply:
👏